packages/material-ui-styles/test/index.spec.tsx

```diff
@@ -99,6 +99,32 @@ function testGetThemeProps(theme: Theme, props: AppBarProps): void {
     // but this would pass anyway
     const alsoClasses = useUnsafeProps(undefined); // runtime: Can't read property color of undefined
   };
+
+  // default theme
+  interface CustomTheme {
+    attribute: number;
+    [key: string]: any;
+  }
+
+  const validCustomTheme = { attribute: 8, otherStuff: true };
+  const invalidCustomTheme = { otherStuff: true };
+
+  const style = createStyles((theme: CustomTheme) => ({
+    margin: theme.attribute,
+  }));
+
+  makeStyles<typeof style, CustomTheme>(style, {
+    defaultTheme: validCustomTheme,
+  });
+
+  makeStyles(style, {
+    defaultTheme: validCustomTheme,
+  });
+
+  // $ExpectError
+  makeStyles<typeof style, CustomTheme>(style, {
```

Comment:
Does this also throw if the generic arguments are inferred?

Reply:
it only throws if theme interfaces mismatch

Reply:
Could you add the test for that too? I much rather test with type inference by default. Simply because I want to document which types work with type inference and which require explicit generic arguments.

Reply:
not sure what you mean this time

Reply:
But not the error case as far as I can tell.

Reply:
which error case do you have in mind? I don't quite get

Reply:
ok, git it. will have a look

Reply:
> How can inference be improved?

Not sure. Sometimes it's just typescript that bails out. 

I'm ok with documenting in the code that this is undesired and follow up with an example that works around this issue (here explicit generic arguments). 

Maybe someone else will later discover a solution and fix it.

Reply:
documented in test